### PR TITLE
cors: Fix handling of multiple allowed origins

### LINKED ISSUE #255: CORS: Multiple origins for access-control-allow-origin when using `AllowOrigin::list`
### Version

0.3.0

### Description

I tried this code:

```rs
CorsLayer::new().allow_methods(Any).allow_origin(AllowOrigin::list([
    "http://localhost:3000",
    "http://192.168.1.10:3000",
]));
```

Send requests from origin `http://localhost:3000`.

I expected to see this happen:
```http
access-control-allow-origin: http://localhost:3000
```

Instead, this happened:
```http
access-control-allow-origin: http://localhost:3000,http://192.168.1.10:3000
```

According to [Access-Control-Allow-Origin](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Access-Control-Allow-Origin), there should not be multiple origins.

### PR DESCRIPTION
**Pull request by jplatte**
Fixes #255.

### CODE REVIEW

File: tower-http/src/cors/allow_origin.rs [hunk 1]
```diff
@@ -129,14 +129,15 @@ impl From<Vec<HeaderValue>> for AllowOrigin {
 
 #[derive(Clone)]
 enum OriginInner {
-    Const(Option<HeaderValue>),
+    Const(HeaderValue),
+    List(Vec<HeaderValue>),
```
Comment (davidpdrsn) [inline tower-http/src/cors/allow_origin.rs:133]: Maybe

```suggestion
    List(Arc<[HeaderValue]>),
```

**Review by davidpdrsn — APPROVED** [review]

File: tower-http/src/cors/allow_origin.rs (hunk 1 above)
Comment (jplatte) [reply tower-http/src/cors/allow_origin.rs:133]: Why? Do you expect the layer to be cloned commonly?

Comment (davidpdrsn) [reply tower-http/src/cors/allow_origin.rs:133]: The layer no but the service contains a layer, so yes.

Comment (jplatte) [reply tower-http/src/cors/allow_origin.rs:133]: And the service will be cloned a lot? Why? Also maybe we should have a single `Arc` around the current fields of the `Layer` then? This seems unrelated in any case.

Comment (davidpdrsn) [reply tower-http/src/cors/allow_origin.rs:133]: Yes services are often cloned to get an owned value that can be moved into the response future. It doesn't have to happen in this PR but is something to keep in mind.

**Pull request merged**